### PR TITLE
concat now throws when concatenating array with non-array

### PR DESCRIPTION
**Pull request by Bradcomp**
Fix for https://github.com/ramda/ramda/issues/1755

### CODE REVIEW

File: src/concat.js [hunk 1]
```diff
@@ -23,4 +23,9 @@ var invoker = require('./invoker');
  *      R.concat([4, 5, 6], [1, 2, 3]); //=> [4, 5, 6, 1, 2, 3]
  *      R.concat('ABC', 'DEF'); // 'ABCDEF'
  */
-module.exports = flip(invoker(1, 'concat'));
+module.exports = _curry2(function concat(a, b) {
+  if (_isArray(a) && !_isArray(b)) {
+    throw new TypeError(toString(b) + ' is not an array');
```
Comment (scott-christopher) [inline src/concat.js]: It looks like we're missing an import of `toString` in here.

Comment (Bradcomp) [inline src/concat.js]: Right you are!

**Comment by scott-christopher** [comment]
Thanks Brad. This looks good to me. 🍄

I'll hold off to give the others a chance to comment before merging.

**Comment by buzzdecafe** [comment]
this is a step in the right direction, because it protects us from the polymorphic native JS concat behavior. 

Should we update the docs to notify that the semantics of `R.concat` are different from `Array.prototype.concat`? 

Otherwise :cow2:

**Comment by Bradcomp** [comment]
Would something like this work (based on note for reduce)? 
"Note: `R.concat` expects both arguments to be of the same type, unlike the native `Array.prototype.concat` method.  It will throw an error if you `concat` an Array with a non-Array value."

**Comment by buzzdecafe** [comment]
LGTM @Bradcomp thx

**Comment by CrossEye** [comment]
:herb:

**Pull request merged**